webapp/src/auth.ts

```diff
@@ -76,23 +79,21 @@ export async function loginAsync(idp: pxt.IdentityProviderId, persistent: boolea
 
     pxt.tickEvent('auth.login.start', { 'provider': idp });
 
-    const stateKey = ts.pxtc.Util.guidGen();
     const stateObj: AuthState = {
-        continuationState,
+        continuationHash,
```

Comment:
I'm curious, why remove the generality of arbitrary state and restrict it to just a hash?

Reply:
Oh I meant to switch this back to `continuationState`. The hash thing was an experiment in making the callback work. Not using hashes anymore. Good catch.

Reply:
I updated this to `callbackLocation`. It can be any url fragment, e.g.: `#editor`. It can contain url parameters too, if needed. This should be adequate for the app to pick up where it left off after sign in is complete. This isn't state sent to the server. It is stored locally and read back when the auth flow completes.